### PR TITLE
update specification to contain clauses & annexes

### PR DESCRIPTION
**Pull request by kestewart**
As part of getting the specification ready to be submitted to ISO,  need to change from sections to clauses, and appendices to annexes.   This commit sets up the order to read the documents in.

### CODE REVIEW

**Comment by RexJaeschke** [comment]
The text covering the Foreword has that word misspelled, twice. It should be

> 'Foreword': foreword.md

Also, at the very end there should be

> 'Bibliography': bibliography.md

**Comment by kestewart** [comment]
Thanks @RexJaeschke - can you check I've got it right now?

File: mkdocs.yml [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
-- 'Appendix I: SPDX License List': appendix-I-SPDX-license-list.md
-- 'Appendix II: License Matching Guidelines and Templates': appendix-II-license-matching-guidelines-and-templates.md
-- 'Appendix III: RDF Data Model Implementation and Identifier Syntax': appendix-III-RDF-data-model-implementation-and-identifier-syntax.md
-- 'Appendix IV: SPDX License Expressions': appendix-IV-SPDX-license-expressions.md
-- 'Appendix V: Using SPDX short identifiers in Source Files': appendix-V-using-SPDX-short-identifiers-in-source-files.md
-- 'Appendix VI: External Repository Identifiers': appendix-VI-external-repository-identifiers.md
-- 'Appendix VII: Creative Commons Attribution License 3.0 Unported': appendix-VII-creative-commons-attribution-license-3.0-unported.md
-- 'Appendix VIII: SPDX Lite': appendix-VIII-SPDX-Lite.md
-- 'Appendix IX: SPDX File Tags': appendix-IX-file-tags.md
+- 'Foreward': foreward.md
```
Comment (swinslow) [inline mkdocs.yml:9]: I think the "a" should be "o" in the title and the filename, so this should be:

```
- 'Foreword': foreword.md
```

File: mkdocs.yml [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
+- 'Introduction': introduction.md
+- 'Clause 1: Scope': scope.md
+- 'Clause 2: Normative references': normative-references.md
+- 'Clause 3: Terms and definitions': terms-and-definitions.md
+- 'Clause 4: Conformance': conformance.md
+- 'Clause 5: Composition of an SPDX document': composition-of-an-SPDX-document.nmd
```
Comment (swinslow) [inline mkdocs.yml:15]: stray "n" in the file extension, ".nmd" should just be ".md"

**Comment by RexJaeschke** [comment]
@kestewart  you are halfway there. It's "foreword", *not* "foreward," and, Yes, the directional word "for[e]ward" has US/British spellings, but that's an entirely different word.

**Comment by kestewart** [comment]
@RexJaeschke - there's a reason I was a math major....   I think that Steve fixed it, and at least now it matches what you intended.    Will apply and cross fingers we've caught all the typos, otherwise, another shorter patch.

**Pull request merged**